Makefile

```diff
@@ -134,15 +134,15 @@ test-unit-race: test-unit ##@tests Run unit and integration tests with -race fla
 test-e2e: ##@tests Run e2e tests
 	# order: reliability then alphabetical
 	# TODO(tiabc): make a single command out of them adding `-p 1` flag.
-	go test -timeout 5m ./t/e2e/accounts/... -network=$(networkid) $(gotest_extraflags)
-	go test -timeout 5m ./t/e2e/api/... -network=$(networkid) $(gotest_extraflags)
-	go test -timeout 5m ./t/e2e/node/... -network=$(networkid) $(gotest_extraflags)
-	go test -timeout 50m ./t/e2e/jail/... -network=$(networkid) $(gotest_extraflags)
-	go test -timeout 20m ./t/e2e/rpc/... -network=$(networkid) $(gotest_extraflags)
-	go test -timeout 20m ./t/e2e/whisper/... -network=$(networkid) $(gotest_extraflags)
-	go test -timeout 10m ./t/e2e/transactions/... -network=$(networkid) $(gotest_extraflags)
+	go test -timeout 5m ./t/e2e/accounts/... -network=$(networkid) -transactions=true $(gotest_extraflags)
```

Comment:
I am not sure what's the purpose of `-transactions` flag. Shouldn't we put checks in individual tests or suites that deals with sending transactions and skip them if run in mainnet?

Reply:
The idea behind the check is to block all test suites doing transaction when testing on mainnet. When doing those tests on other networks or they don't contains tests with transactions they won't block.

Reply:
So, here `./t/e2e/accounts/...` and `-transactions=true` mean that we marked these group of tests as dangerous and panic if they run on mainnet? 

I don't see how it can be useful as in a single test we can do something like this:
```go
func TestSendingTx(t *testing.T) {
  if GetNetworkID() == params.MainnetNetworkID {
    t.Skip()
  }
}
```

Generally, there should be a pattern when `GetNetworkID()` returns mainnet or our private chain and then such a test should be skipped.

Reply:
So war we blocked all mainnet tests with a panic. I this approach tries to reduce this to only transactional suites. Sure can do simple skips for individual tests, sadly AFAIK not with `Skip()` but `return`.